app/src/main/java/org/mozilla/fenix/library/history/HistoryAdapter.kt

```diff
@@ -16,6 +17,7 @@ import org.mozilla.fenix.library.history.viewholders.HistoryListItemViewHolder
  */
 class HistoryAdapter(
     private val historyInteractor: HistoryInteractor,
+    private val syncedHistoryVisible: Boolean,
```

Comment:
I don't think we need to pass this here. We can call `FeatureFlags.syncedHistoryFeature` from anywhere so we don't need to pass this around like this.

Reply:
I would actually prefer passing a parameter rather than accessing a static class within a UI class.
Giving the UI class knowledge about `FeatureFlags` is not the clearest solution.
Also passing parameters, that affect the behavior of a class, inside the constructor makes it easier to understand. We can see the params in charge, where they are used, and can track where they come from.

Reply:
We don't expect this parameter to be around when the `FeatureFlag` is enabled. When this is enabled, it would be easier to remove your single direct check of the `FeatureFlag` instead of the refactoring out `syncedHistoryVisible`.

Reply:
I think simple is best here. I can't pass a solution where we end up deciding to not document our public class/functions after promoting the need for more documentations. The refactoring is also a bit drastic because of the fact that we can call it the FeatureFlag statically and avoid a lot of refactoring altogether. Alternatively, we could've added it as a an extra parameter and suppress LongParameterList. My other concern is that this refactoring also affects the git blame/history of code where it is probably more valuable to preserve.